### PR TITLE
Benji/finalizingonboarding

### PR DESCRIPTION
**Pull request by Benjamindodgson**
The goal here was to properly handle user state in the backend based on things like reservationId/pass/user status. Previously most things were handled in validate code, before we had a name or notification permissions for a user. Now all that logic is moved into the updated FinalizeOnboarding call. 

Waiting on a few bugs to be addressed before merging.

### CODE REVIEW

File: Shared/API/Onboarding+Cloud.swift [hunk 1]
```diff
@@ -42,16 +42,12 @@ struct VerifyCode: CloudFunction {
     let code: String
     let phoneNumber: PhoneNumber
     let installationId: String
-    let reservationId: String
```
Comment (Benjamindodgson) [inline Shared/API/Onboarding+Cloud.swift:45]: removing this as this call no longer handles these properties.

File: Shared/API/Onboarding+Cloud.swift [hunk 2]
```diff
@@ -73,16 +69,22 @@ struct VerifyCode: CloudFunction {
     }
 }
 
-struct ActivateUser: CloudFunction {
+struct FinalizeOnboarding: CloudFunction {
```
Comment (Benjamindodgson) [inline Shared/API/Onboarding+Cloud.swift:72]: More accurate name

File: Shared/Models/Parse Subclasses/Connection.swift [hunk 3]
```diff
@@ -13,8 +13,6 @@ enum ConnectionKey: String {
     case status
     case to
     case from
-    case conversationSid
-    case initialConversations
```
Comment (Benjamindodgson) [inline Shared/Models/Parse Subclasses/Connection.swift:17]: No longer used.

File: Shared/Stores/UserStore.swift [hunk 4]
```diff
@@ -94,9 +94,15 @@ class UserStore {
     }
     
     func mapMembersToUsers(members: [ConversationMember]) async throws -> [User] {
-        return try await members.userIDs.concurrentMap { userId in
-            return await self.findUser(with: userId)!
+        
+        var users: [User] = []
+        await members.userIDs.asyncForEach { userId in
```
Comment (Benjamindodgson) [inline Shared/Stores/UserStore.swift:99]: fixes crash

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 5]
```diff
@@ -216,6 +216,16 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
             self.collectionView.setContentOffset(CGPoint(x: 0, y: yOffset), animated: true)
 
             await Task.sleep(seconds: Theme.animationDurationStandard)
+            
+            if animateSelection, let cell = self.collectionView.cellForItem(at: messageIndexPath) {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:220]: little pop to message when deep linking

File: iOS/Flows/Conversation/ConversationListCoordinator.swift [hunk 6]
```diff
@@ -83,21 +83,8 @@ class ConversationListCoordinator: PresentableCoordinator<Void>, ActiveConversat
 
         switch target {
         case .conversation:
-            var cid: ConversationId?
             let messageID = deeplink.messageId
-    
-            if deeplink.conversationId.exists {
-                cid = deeplink.conversationId
-
-            } else if let connectionId = deeplink.customMetadata["connectionId"] as? String {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationListCoordinator.swift:92]: we no longer include a connection in the payload of a notification.

File: Shared/Flows/Onboarding/OnboardingCoordinator.swift [hunk 7]
```diff
@@ -178,25 +178,22 @@ extension OnboardingCoordinator: OnboardingViewControllerDelegate {
...
+            case .waitlist, .inactive, .active:
+                self.finalizeOnboarding(user: user)
             }
         }
     }
     
-    func activateUserThenShowPermissions(user: User) {
+    func finalizeOnboarding(user: User) {
         Task {
+            await self.checkForPermissions()
```
Comment (Benjamindodgson) [inline Shared/Flows/Onboarding/OnboardingCoordinator.swift:189]: check for permissions before finalizing

File: Shared/Models/AnimationCycle.swift [hunk 8]
```diff
@@ -9,8 +9,8 @@
 import Foundation
 
 struct AnimationCycle {
-    var inFromPosition: AnimationPosition
-    var outToPosition: AnimationPosition
+    var inFromPosition: AnimationPosition?
```
Comment (Benjamindodgson) [inline Shared/Models/AnimationCycle.swift:12]: Making this optional for future work.

File: Shared/UI/Face Detection/FaceDetectionViewController+Extensions.swift [hunk 9]
```diff
@@ -38,7 +38,7 @@ extension FaceDetectionViewController {
         let color = ThemeColor.B3.ciColor
         
         // Create CIImage objects for the video frame and the segmentation mask.
-        let originalImage = CIImage(cvPixelBuffer: framePixelBuffer).oriented(.leftMirrored)
+        let originalImage = CIImage(cvPixelBuffer: framePixelBuffer).oriented(self.orientation)
         var maskImage = CIImage(cvPixelBuffer: maskPixelBuffer)
```
Comment (Benjamindodgson) [inline Shared/UI/Face Detection/FaceDetectionViewController+Extensions.swift:42]: addressing scale and orientation issues.

**Review by MartinJibber — APPROVED** [review]

File: iOS/Flows/Conversation/Conversation Header/MembersCollectionViewLayout.swift [hunk 10]
```diff
@@ -51,42 +51,41 @@ class MembersCollectionViewLayout: UICollectionViewCompositionalLayout {
...
-        #warning("fix layout")
-        return super.layoutAttributesForElements(in: rect)
-//        guard let attributes = super.layoutAttributesForElements(in: rect) as? [MemberCellLayoutAttributes] else {
-//            return nil
-//        }
-//        return attributes.map({ attribute in
-//            let copy = attribute.copy() as! MemberCellLayoutAttributes
-//            return self.transformLayoutAttributes(copy)
-//        })
+        guard let attributes = super.layoutAttributesForElements(in: rect) as? [MemberCellLayoutAttributes] else {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Header/MembersCollectionViewLayout.swift:54]: Almost fixed this! it basically works but I couldn't get it to animate in and out correctly.

**Pull request merged**